Build updates

bsamartins opened the pull request:
Build updates and changes for #3:
- Migrate gradle build scripts to Kotlin
- Update JUnit to JUnit Jupiter
- Target Java8

bsamartins committed:
Migrate gradle build scripts to Kotlin

bsamartins committed:
Refactor sourceJar task to task closure

bsamartins committed:
Migrate tests to JUnit Jupiter

bsamartins committed:
Target Java 8

bsamartins committed:
Replace oraclejdk8 with openjdk

bsamartins commented:
Updated build jdk from oraclejdk to openjdk as it couldn't be installed

holgerstolzenberg commented on build.gradle.kts:
```diff
@@ -0,0 +1,95 @@
+dependencies {
+  val junitVersion: String by project
+
+  implementation(platform("org.junit:junit-bom:$junitVersion"))
```
- Why isn´t this `testImplementation` scope also. Are there any JUnit dependencies needed in the final artifact?
- Can the JUnit version be inlined to the dependency declaration?

bsamartins replied:
I've inlined the junit version. Since junit dependencies are managed by that bom, it made the version in gradle.properties not so relevant.

As for the bom import, in this case doesn't really matter whether it sits in the implementation or testImplementation configuration (unless you wanted a junit dependency from being added by mistake to implementation).

Declaring the platform definitions `implementation(platform("org.junit:junit-bom:$junitVersion"))` does not necessarily add those dependencies to the project, it only tells gradle how to resolve the constraints declared by that BOM (similar to what happens with the Maven dependency management).

The actually dependencies are not included in the project until they're added in the dependencies block:

```
dependencies {
  ...
   testImplementation("org.junit.jupiter:junit-jupiter-api")
   testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine")
 ...
}
```

I've added the BOM import to the test implementation as it shouldn't constraint junit dependencies but in the test implementation.

holgerstolzenberg replied:
👍🏾

holgerstolzenberg reviewed (commented):
Back from holidays 😎I have just one question regarding the PR. @bsamartins please comment so that I can finally approve the PR.

bsamartins committed:
Update code review:
- Inline junit version
- Move junit dependency management to test implementation
